### PR TITLE
refactor: refactor Input code

### PR DESCRIPTION
**Pull request by afc163**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [x] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution

优化一下 Input 代码。

### 📝 Changelog

| Language   | Changelog |
| ---------- | ---------- |
| 🇺🇸 English |      --     |
| 🇨🇳 Chinese |      --     |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Review by li-jia-nan — APPROVED** [review]

File: components/input/Input.tsx [hunk 1]
```diff
@@ -172,6 +171,15 @@ const Input = forwardRef<InputRef, InputProps>((props, ref) => {
     </>
   );
 
+  const getAddon = (addon: React.ReactNode) =>
+    addon && (
```
Comment (li-jia-nan) [inline components/input/Input.tsx:175]: 感觉这里没必要写短路了，下面直接写成 `addonAfter={addonAfter ? getAddon(addonAfter) : null}` 好一点

Comment (afc163) [reply components/input/Input.tsx:175]: 为啥这样更好？

**Comment by afc163** [comment]
体积还变大了。。

**Comment by li-jia-nan** [comment]
补充一个：
- #49428

**Pull request merged**